File: src/main/java/org/proofit/premiumcalc/calculation/Premium.java
package org.proofit.premiumcalc.calculation;

import java.math.BigDecimal;

public interface Premium {
    BigDecimal getCoefficient(BigDecimal sumInsured);

    default BigDecimal calculate(BigDecimal sumInsured) {
        return sumInsured.multiply(getCoefficient(sumInsured));
    }
}

File: src/main/kotlin/org/proofit/premiumcalc/calculation/Premium.kt
package org.proofit.premiumcalc.calculation

import java.math.BigDecimal

interface Premium {
    fun getCoefficient(sumInsured: BigDecimal): BigDecimal
    fun calculate(sumInsured: BigDecimal): BigDecimal {
        return sumInsured.multiply(getCoefficient(sumInsured))
    }
}
